lib/bolt/transport/powershell.rb

```diff
@@ -195,7 +195,7 @@ def shell_init
   $propbag = @{}
   foreach ($name in Get-Member -InputObject $xml -MemberType Properties | Where-Object{$_.Name -notmatch "^__|\\btype\\b"} | Select-Object -ExpandProperty name) {
```

Comment:
doh, i need to delete the regex part.

Reply:
Can we just change the regex from `"^__|type"` to `"^(__|type)"`?

Reply:
I think we want to avoid the regex because it seems like there is always a key with the `name` `type` and it's value `object` (not sure how to dig deeper in to what *kind* of object). However, each of the `$xml.$Name` objects we want always have the type `[Xml.XmlElement]`. So the idea is to filter out the parameters prepended with underscores and then add only those key/value to `propbag` when the value is an xml element.

Reply:
Actually, `type` is still lost with this approach. Somewhere the key `type` gets overridden. @Iristyle Do you see a way around that or should we just update the regex and then document that powershell parameters cannot be named `type` with powershell 2?

Reply:
I'm not sure I'm totally following, but I don't think you want to remove the omission of `type` here.

Reply:
So, any parameter that is *not* called `type` is a System.Xml.XmlElement. For some reason parameters with the name type are 

If you modify the following debug line 
https://github.com/puppetlabs/bolt/blob/49cec2cd552519f278bc843aaa763766c5a3905c/lib/bolt/transport/powershell.rb#L197 to print out
`"$Name Type: $($xml.$Name)"`
The type parameter appears like:
`type Type: object type`
Where any other parameter (for example `types`) appears as expected:
`types Type: System.Xml.XmlElement`
What is curious is that when there is no parameter named `type` there is always an entry for:
`type Type: object`
So my thought is, that somehow the `type` parameter (when specified in powershell sig) is getting overwritten/merged with the inherent `$xml.$Name` `type` that is  always present.

Reply:
With this implementation parameters with the name `type` will be clobbered. However, updating the regex allows parameters that contain the string `type`.